### PR TITLE
Add createFile param to the switch submode command

### PR DESCRIPTION
**Pull request by jurgenwerk**
This PR addresses a user experience issue where:

1. User is in interact mode and opens the AI assistant and tells it to generate something (e.g. "make me a tic tac toe game")
2. The assistant will switch to code mode and open the index card, then proceed with ai generation and once user accepts the patch to create file(s) nothing happens, they still see the index card.

This PR adds an optional boolean param `createFile` to switch mode command where it will create the file the assistant is about to generate. The user will be transitioned to code mode, seeing a blank file until the AI assistant is done generating the code.

The changes in this PR alone don't fix the mentioned issue yet - there will be another PR in the skills repo to teach the LLM when to use `createFile` parameter (I have it almost ready and confirm it works with all of the above).

---

> [!NOTE]
> Introduces optional file creation when switching to code mode and centralized non-conflicting filename handling.
> 
> - Adds `createFile` to `SwitchSubmodeInput` and new `SwitchSubmodeResult` (returns `codePath` when filename changes)
> - Updates `switch-submode` to optionally create a blank file via `write-text-file` when entering "code" from "interact"; updates code path accordingly
> - Enhances `write-text-file` with `useNonConflictingFilename`, realm validation, safer existence checks, and returns `FileUrlCard`
> - Extracts `findNonConflictingFilename` to `utils/file-name` and refactors `patch-code` to use it
> - UI: suppresses result card display for `switch-submode*` commands in matrix message renderer
> - Adds comprehensive tests for new behaviors and edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 292028492af4b894f11d5f39b2f38d08fe002fce. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by jurgenwerk** [comment]
Accompanying PR: https://github.com/cardstack/boxel-skills/pull/44

File: packages/host/app/commands/switch-submode.ts [hunk 1]
```diff
@@ -43,24 +48,55 @@ export default class SwitchSubmodeCommand extends HostBaseCommand<
...
+          } else if (status === 200) {
+            if (content.trim() !== '') {
+              let nonConflictingUrl = await findNonConflictingFilename(
+                codeUrl.href,
+                (candidateUrl) => this.fileExists(candidateUrl),
+              );
+              let newCodeUrl = new URL(nonConflictingUrl);
+              await this.cardService.saveSource(newCodeUrl, '', 'create-file');
+              finalCodeUrl = newCodeUrl;
+
```
Comment (lukemelia) [inline packages/host/app/commands/switch-submode.ts]: It might be elegant to delegate to a separate command to accomplish this.

File: packages/runtime-common/ai/prompt.ts [hunk 2]
```diff
@@ -809,6 +809,13 @@ async function toResultMessages(
         } else {
           content = `Tool call ${status == 'applied' ? 'executed' : status}.\n`;
         }
+        let switchSubmodeInstruction =
+          getSwitchSubmodeInstruction(commandResult);
+        if (switchSubmodeInstruction) {
+          content = [content, switchSubmodeInstruction]
+            .filter(Boolean)
+            .join('\n\n');
+        }
```
Comment (lukemelia) [inline packages/runtime-common/ai/prompt.ts]: I'm not enthusiastic about having command-specific code in the ai prompt generation. Is this necessary?

File: packages/host/app/commands/switch-submode.ts (hunk 1 above)
Comment (jurgenwerk) [reply packages/host/app/commands/switch-submode.ts]: I'll check if I can use `WriteTextFileCommand` for this

File: packages/runtime-common/ai/prompt.ts (hunk 2 above)
Comment (jurgenwerk) [reply packages/runtime-common/ai/prompt.ts]: My intent was to not add more text to the skills prompt since this should be a rare case. I checked and we can achieve this by adding a line to the development skill and drop this new logic in the ai bot

File: packages/host/app/commands/write-text-file.ts [hunk 3]
```diff
@@ -26,40 +29,69 @@ export default class WriteTextFileCommand extends HostBaseCommand<
...
+          shouldWrite = true;
+        } else if (existing.status === 200) {
+          if (existing.content.trim() !== '') {
+            let nonConflictingUrl = await findNonConflictingFilename(
+              url.href,
+              (candidateUrl) => this.fileExists(candidateUrl),
+            );
+            finalUrl = new URL(nonConflictingUrl);
+          } else {
+            shouldWrite = false;
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/host/app/commands/write-text-file.ts]: Fixed

File: packages/host/app/commands/switch-submode.ts (hunk 1 above)
Comment (jurgenwerk) [reply packages/host/app/commands/switch-submode.ts]: I adjusted WriteTextFileCommand to fit this case and used it in this command

File: packages/runtime-common/ai/prompt.ts (hunk 2 above)
Comment (jurgenwerk) [reply packages/runtime-common/ai/prompt.ts]: Ai bot logic removed, and I added some extra instructions in the development skill: https://github.com/cardstack/boxel-skills/pull/44/changes/27195755660f0badb1ec6f94609e40a5ea401017#r2721123671

File: packages/base/command.gts [hunk 4]
```diff
@@ -117,13 +117,19 @@ export class FileContents extends CardDef {
...
+export class SwitchSubmodeResult extends CardDef {
+  @field codePath = contains(StringField);
 }
 
 export class WriteTextFileInput extends CardDef {
   @field content = contains(StringField);
   @field realm = contains(StringField);
   @field path = contains(StringField);
   @field overwrite = contains(BooleanField);
+  @field useNonConflictingFilename = contains(BooleanField);
```
Comment (lukemelia) [inline packages/base/command.gts:132]: Perhaps it should be an error in the command if the caller passes both overwrite and useNonConflictingFilename as true?

**Review by lukemelia — APPROVED** [review]

File: packages/base/command.gts (hunk 4 above)
Comment (jurgenwerk) [reply packages/base/command.gts:132]: 👍 good idea
